book/machinelearning.qmd

```diff
@@ -4,114 +4,215 @@ abstract: TODO (150-200 WORDS)
+If a model were to be trained and evaluated on the same data, the resulting loss, known as the *training error*, would be an overoptimistic estimate of the true generalization error (@Hastie2013) as the model is making predictions for data it has already 'seen' and the loss is not evaluating its ability to generalize to new, unseen data.
+Estimation of the generalization error requires *data splitting*, which is the process of splitting available data, $\calD$, into *training data*, $\dtrain \subset \calD$, and *testing data*, $\dtest = \calD \setminus \dtrain$.
+<!-- FIXME: ABOVE COULD BE BETTER WORDED -->
+
+The simplest method to estimate the generalization error is to use *holdout resampling*, which is the process of partitioning the data into one training dataset and one testing dataset, with the model trained on the former and predictions made for the latter.
+Using 2/3 of the data for training and 1/3 for testing is a common splitting ratio (@Kohavi1995).
+When splitting, it is essential the data is partitioned randomly to ensure any information encoded in data ordering is removed.
+Ordering is often important in the real-world, for example in healthcare data when patients are recorded in order of enrolment to a study.
+Whilst ordering can provide useful information, it does not generalize to new, unseen data.
+For example, the number of days a patient has been in hospital is more useful than the patient's index in the dataset, as the former could be calculated for a new patient whereas the latter is meaningless.
+Another example is in the `rats` dataset that was briefly described in @sec-surv.
+The `rats` data explores how a novel drug effects tumor incidence.
+However, the data is ordered by rat litters with every three rats being in the same litter.
+Hence if rats were to be bred or raised differently over time,  even if the `litter` column were removed, this information would still be encoded in the order of the dataset and could impact upon any findings.
+Randomly splitting breaks any possible association between order and outcome.
+<!-- FIXME: ABOVE IS ALSO BAD! -->
+
+Holdout resampling is a quick method to estimate the generalization error, however it is an imperfect method as, even with random splitting, it can still result in 'unlucky' splits.
+Returning to the `rats` data, consider a split that results in only male rats in the training data, the model would then be unable to handle female rats in the testing data, which it had never encountered before.
+Alternatively, consider a random split with an even sex balance, as only two out of the 150 rats have a tumor, there is a high probability that the training data will not include any male rat with a tumor.
+Hence, in the case of a male rat, the model is likely to ignore the treatment variable and instead just use sex as the most important predictor of tumor incidence.
```

Comment:
The reason to use CV is because hold-out has high variance for low-dim data. The problems mentioned here could be solfed by stratified resampling